### PR TITLE
Deprecate `MergeNQubitGates` optimizer 

### PR DESCRIPTION
**Pull request by tanujkhattar**
Part of https://github.com/quantumlib/Cirq/issues/5028

### CODE REVIEW

File: cirq-core/cirq/contrib/quimb/grid_circuits.py [hunk 1]
```diff
@@ -89,6 +89,7 @@ def _interaction(
     )
 
 
+@cirq._compat.deprecated_class(deadline='v0.16', fix="Use cirq.merge_k_qubit_unitaries")
 class MergeNQubitGates(cirq.PointOptimizer):
```
Comment (vtomole) [inline cirq-core/cirq/contrib/quimb/grid_circuits.py:93]: This class does more than `cirq.merge_k_qubit_unitaries`. It calls off to `cirq.drop_negligible_operations` and `cirq.drop_empty_moments` as well. Will existing tests for this optimizer pass if only `cirq.merge_k_qubit_unitaries` is called?

Comment (tanujkhattar) [reply cirq-core/cirq/contrib/quimb/grid_circuits.py:93]: Can you point me to where `MergeNQubitGates` optimizer calls off to `cirq.drop_negligible_operations` and `cirq.drop_empty_moments` ? 

Also, the call sites calling `MergeNQubitGates` have been updated to call `cirq.merge_k_qubit_unitaries` and no tests fail right now. But to answer your specific question, `cirq.merge_k_qubit_unitaries` does not call `cirq.drop_negligible_operations` and `cirq.drop_empty_moments` by default.

Comment (vtomole) [reply cirq-core/cirq/contrib/quimb/grid_circuits.py:93]: Whoops, i mistook this for `simplify_expectation_value_circuit` before.

**Review by vtomole — APPROVED** [review]
LGTM

**Pull request merged**